watch/plan.go

```diff
@@ -57,6 +57,7 @@ OUTER:
 		if err != nil {
 			// Perform an exponential backoff
 			failures++
+			p.lastIndex = 1
```

Comment:
I don't think this'll catch all cases since you won't always get an error. It's probably better to add something down on line 88 like this:

```
if p.lastIndex < oldIndex {
    p.lastIndex = 0
}
```

This'll reset the blocking query whenever the index slips backwards.